client/src/pages/Login/Login.tsx

```diff
@@ -5,23 +5,30 @@ import login from '../../helpers/APICalls/login';
 import LoginForm from './LoginForm/LoginForm';
 import { useAuth } from '../../context/useAuthContext';
 import { useSnackBar } from '../../context/useSnackbarContext';
+import { getUProfile } from '../../helpers/APICalls/getUProfile';
 
 export default function Login(): JSX.Element {
   const history = useHistory();
 
-  const { updateLoginContext } = useAuth();
+  const { updateLoginContext, updateProfileContext } = useAuth();
   const { updateSnackBarMessage } = useSnackBar();
 
   const handleSubmit = (
     { email, password }: { email: string; password: string },
     { setSubmitting }: FormikHelpers<{ email: string; password: string }>,
   ) => {
-    login(email, password).then((data) => {
+    login(email, password).then(async (data) => {
       if (data.error) {
         setSubmitting(false);
         updateSnackBarMessage(data.error.message);
       } else if (data.success) {
-        updateLoginContext(data.success);
+        const user = data.success;
+        await getUProfile().then((res) => {
+          if (res.success) {
+            updateProfileContext(res.success);
+            updateLoginContext(user);
+          }
+        });
```

Comment:
I'm once again concerned about not letting the user sign in if they don't have a profile (I'm not sure when a profile is created in your implementation, but in any case something seems off about refusing to log in a user because we can't find a profile). I'd like to hear your thoughts about this 🤔

Reply:
> I'm once again concerned about not letting the user sign in if they don't have a profile (I'm not sure when a profile is created in your implementation, but in any case something seems off about refusing to log in a user because we can't find a profile). I'd like to hear your thoughts about this 🤔

Yeah you are right, should be outside success check. SO no matter the getUprofile function works, the user updated after the get user profile try.